### PR TITLE
Add unit test for MaskDesignerDialog

### PR DESCRIPTION
**Pull request by Liv-Goh**
Related https://github.com/dotnet/winforms/issues/10773

## Proposed changes
1. Add unit test MaskDesignerDialogTests.cs for public properties and method of the MaskDesignerDialog.
2. Enable nullability in MaskDesignerDialogTests..cs.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12575)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+namespace System.Windows.Forms.Design.Tests;
+
+public sealed class MaskDesignerDialogTests:IDisposable
+{
+    private readonly MaskedTextBox _maskedTextBox;
+    private readonly MaskDesignerDialog _dialog;
+
+    public MaskDesignerDialogTests()
+    {
+        _maskedTextBox = new MaskedTextBox();
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:18]: You can omit the type in the `new()` instance creations inside this constructor.

Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:18]: You can omit the type in the `new()` instance creations inside this constructor.

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    public void Dispose()
+    {
+        _maskedTextBox.Dispose();
+        _dialog.Dispose();
+    }
+
+    [Fact]
+    public void Constructor_ValidMaskedTextBox_UsesProvidedMaskedTextBox()
+    {
+        var txtBoxMask = (TextBox)_dialog.TestAccessor().Dynamic._txtBoxMask;
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:31]: Please declare types explicitly. This happens in other areas of the code as well. 

Also, please add a `using` statement to dispose of the TextBox.

Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:31]: Please declare types explicitly. This happens in other areas of the code as well.

**Review by ricardobossan — COMMENTED** [review]
Other than small comments, LGTM!

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        _dialog.ValidatingType.Should().Be(typeof(DateTime));
+    }
+
+    [Fact]
+    public void MaskDescriptorsEnumerator_ShouldReturnCorrectDescriptors()
+    {
+        Collections.IEnumerator enumerator = _dialog.MaskDescriptors;
+
+        List<MaskDescriptor> descriptors = new List<MaskDescriptor>();
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:50]: Please use target type new
```suggestion
        List<MaskDescriptor> descriptors = new();
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    [WinFormsTheory]
+    [InlineData(typeof(ValidMaskDescriptor), true)]
+    [InlineData(typeof(AbstractMaskDescriptor), false)]
+    [InlineData(typeof(NonPublicMaskDescriptor), false)]
+    public void DiscoverMaskDescriptors_ShouldHandleVariousDescriptorTypes(Type descriptorType, bool shouldBeAdded)
+    {
+        Mock<ITypeDiscoveryService> mockDiscoveryService = new Mock<ITypeDiscoveryService>();
+        List<Type> types = new List<Type> { descriptorType };
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:78]:
```suggestion
        Mock<ITypeDiscoveryService> mockDiscoveryService = new();
        List<Type> types = new() { descriptorType };
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+#nullable enable
+
+using System.ComponentModel.Design;
+using Moq;
+
+namespace System.Windows.Forms.Design.Tests;
+
+public sealed class MaskDesignerDialogTests : IDisposable
+{
+    private readonly MaskedTextBox _maskedTextBox = new MaskedTextBox();
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:13]: Same here

**Review by lonitra — COMMENTED** [review]
Minor comments about using target type new otherwise 👍

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs [hunk 6]
```diff
@@ -28,7 +27,7 @@ public void Dispose()
     [Fact]
     public void Constructor_ValidMaskedTextBox_UsesProvidedMaskedTextBox()
     {
-        var txtBoxMask = (TextBox)_dialog.TestAccessor().Dynamic._txtBoxMask;
+        TextBox txtBoxMask = _dialog.TestAccessor().Dynamic._txtBoxMask;
```
Comment (ricardobossan) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs:30]: Please add a `using` statement here, to dispose of the control.

**Review by ricardobossan — APPROVED** [review]

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**